src/pybind/mgr/volumes/fs/volume.py

```diff
@@ -595,13 +597,21 @@ def create_subvolume_group(self, **kwargs):
         try:
             with open_volume(self, volname) as fs_handle:
                 try:
-                    with open_group(fs_handle, self.volspec, groupname):
+                    with open_group(fs_handle, self.volspec, groupname) as group:
                         # idempotent creation -- valid.
-                        pass
+                        attrs = {
+                            'uid': uid if uid is not None else group.uid,
+                            'gid': gid if gid is not None else group.gid,
+                            'mode': octal_str_to_decimal_int(mode) if mode else group.mode,
+                            'data_pool': pool,
+                            'quota': size
```

Comment:
Also, the idempotent subvolumegroup creation might mess up the quota. Imagine the scenario below.

1. Subvoluemgroup is created with quota
2. Idemptoent subvolumegroup creation is done to set only `uid/gid` without passing `quota`

This would reset the `quota` set earlier to `infinite`. This is incorrect and should pick the existing quota right ?
I think this bug exists in subvolume idempotent creation as well.

Reply:
Ok, I get it. I thought, it should behave as 'update'. Then we might have to fix the subvolume behavior for the 'uid/gid'.